src/components/Loans/OpenLoans/components/BulkClaimReturnedModal/BulkClaimReturnedModal.js

```diff
@@ -0,0 +1,267 @@
+  // with keys present only if the item in question has >0 requests. Thus, if
+  // none of the selected items have any requests on them, requestCounts === {}.
+  getRequestCountForItem = id => this.props.requestCounts[id] || 0;
+
+  handleAdditionalInfoChange = e => {
+    this.setState({ additionalInfo: e.target.value });
+  };
+
+  claimItemReturned = (loan) => {
+    if (!loan) return null;
+
+    this.props.mutator.loanId.replace(loan.id);
+    return this.props.mutator.claimReturned.POST(
+      {
+        itemClaimedReturnedDateTime: moment().format(),
+        comment: this.state.additionalInfo,
+      }
+    );
+  }
+
+  claimAllReturned = () => {
+    this.setState({ operationResults: [] });
+    const promises = Object
+      .values(this.props.checkedLoansIndex)
+      .map(loan => this.claimItemReturned(loan).catch(e => e));
+    Promise.all(promises)
+      .then(results => this.finishClaims(results));
+  }
+
+  finishClaims = (results) => {
+    this.setState({
+      operationResults: results,
+      operationState: 'post',
+    });
+  }
+
+  // @results: array of results of the claim-returned operation for each loan
+  // Each item in the array will either be a simple object that represents a
+  // successful change (including an id value) or a larger object representing
+  // a CORS error (identifiable by the properties ok: false and url).
+  //
+  // @return array of loan IDs for which the operation returned an error.
+  getUnchangedLoansList = (results) => {
+    return results.filter(r => r.ok === false).map(r => r.url?.match(/^.*\/(.*)\/claim-item-returned/)[1]);
+  }
+
+  onCancel = () => {
+    this.setState({ operationState: 'pre' });
+    this.props.onCancel();
+  }
+
+  // Renders a modal dialog containing a list of items to be claimed returned for both
+  // pre-operation review and post-operation feedback (i.e., there's one modal, but the
+  // contents will be different before and after clicking the 'Confirm' button.)
+  render() {
+    const {
+      checkedLoansIndex,
+      intl,
+      open
+    } = this.props;
+    const {
+      additionalInfo,
+      operationResults,
+      operationState,
+    } = this.state;
+    const loans = checkedLoansIndex ? Object.values(checkedLoansIndex) : [];
+    // List of loans for which the claim returned operation failed
+    const unchangedLoans = this.getUnchangedLoansList(operationResults);
+
+    // Controls for the preview dialog
+    const preOpFooter = (
+      <ModalFooter>
+        <Layout className="textRight">
+          <Button
+            data-test-bulk-cr-cancel-button
+            onClick={this.onCancel}
+          >
+            <FormattedMessage id="ui-users.cancel" />
+          </Button>
+          <Button
+            data-test-bulk-cr-confirm-button
+            buttonStyle="primary"
+            onClick={this.claimAllReturned}
+            disabled={!additionalInfo}
+            onChange={this.handleAdditionalInfoChange}
```

Comment:
Sorry, Matt, I can't figure out what the `onChange` property for a button is for.

Reply:
Neither can I! Good catch -- I'm not sure how that got there.